app/scrapers/courses_scraper.rb

```diff
@@ -4,54 +4,70 @@
 require 'open-uri'
 require 'nokogiri'
 
-require_relative 'scraper_common.rb'
+require_relative 'scraper_common'
 include ScraperCommon
 
-require_relative '../models/courses.rb'
+require_relative '../models/courses'
 
-prog_name = "courses_scraper"
+prog_name = 'courses_scraper'
 logger = ScraperCommon::logger
 
 # List semesters in year in testudo's format
 # 2018 -> 201801, 201805, 201808, 201812
+# @type [Array<String>]
 semesters = ScraperCommon::get_semesters(ARGV)
 
-logger.info(prog_name) {semesters}
+logger.info(prog_name) { semesters }
 
 # Get the urls for all the department pages
 dep_urls = []
 semesters.each do |semester|
-  logger.info(prog_name) {"Searching for courses in term #{semester}"}
+  logger.info(prog_name) { "Searching for courses in term #{semester}" }
 
   base_url = "https://app.testudo.umd.edu/soc/#{semester}"
 
-  Nokogiri::HTML(open(base_url)).search('span.prefix-abbrev').each do |e|
+
+  ScraperCommon::get_page(base_url, prog_name).search('span.prefix-abbrev').each do |e|
     dep_urls << "https://app.testudo.umd.edu/soc/#{semester}/#{e.text}"
   end
 
-  logger.info(prog_name) {"#{dep_urls.length} department/semesters so far"}
+  logger.info(prog_name) { "#{dep_urls.length} department/semesters so far" }
 end
 
+##
 # safely formats to UTF-8
-def utf_safe text
-  if !text.valid_encoding?
-    text = text.encode('UTF-8', 'binary', invalid: :replace, undef: :replace, replace: '')
-  end
+#
+# @param [String] text
+# @return [String]
+#
+def utf_safe(text)
+  text = text.encode('UTF-8', 'binary', invalid: :replace, undef: :replace, replace: '') unless text.valid_encoding?
   text
 end
 
 queries = []
+dept_count = Float(dep_urls.length)
```

Comment:
float cast doesn't seem to be necessary, since we cast `i` to a float below? just use `dep_urls.length`

Reply:
yes, but that gets a little messy. and it needs to be casted before division, otherwise truncation occurs

Reply:
wdym? the division should still return the same thing if this is an int, since one of the arguments (`i`) is a float